source/extensions/tracers/zipkin/tracer.cc

```diff
@@ -66,14 +68,25 @@ SpanPtr Tracer::startSpan(const Tracing::Config& config, const std::string& span
   span_ptr->setName(span_name);
 
   // Set the span's kind (client or server)
-  if (config.operationName() == Tracing::OperationName::Egress) {
-    annotation.setValue(CLIENT_SEND);
+  if (split_spans_for_request_) {
+    // If previous context is inner context then this span must be span created for upstream
+    // request. Client span will be created for upstream request when split_spans_for_request is
+    // set to true.
```

Comment:
Can you please clarify this comment?

Reply:
It's a little hard to make it clearer. 🤣 

For a single client request, envoy accept it (downstream request phase), process it (by filters), and send it to server (upstream request phase).

If the `previous_context` is inner context. Then it means the current span is **not first span** for a single request. Because the first span only has external `previous_context` or has no `previous_context`.

If current span is not first span, then this span must be a span that created for an upstream request that created by router and will be send to upstream server.

And if `split_spans_for_request` is set to true, the upstream requests' span should have type of `CLIENT_SEND`.